idea/indexer/src/gear/connect.ts

```diff
@@ -3,41 +3,69 @@ import { RMQServiceActions } from '@gear-js/common';
 
 import config from '../config';
 import { changeStatus } from '../healthcheck';
-import { GenesisCb, logger } from '../common';
+import { GenesisCb, getProviderAddress, logger } from '../common';
 import { GearIndexer } from './indexer';
 
-const MAX_RECONNECTIONS = 10;
+let addresses = config.gear.providerAddresses;
+const MAX_RECONNECTIONS = 10; //max count reconnection for each provider address
 let reconnectionsCounter = 0;
+export let providerAdd = getProviderAddress(addresses);
+let connectionStatus;
 
 export async function connectToNode(indexer: GearIndexer, cb: GenesisCb) {
-  const api = new GearApi({ providerAddress: config.gear.wsProvider });
+  const api = new GearApi({ providerAddress: providerAdd });
+
   try {
     await api.isReadyOrError;
+    connectionStatus = true;
   } catch (error) {
-    logger.error(`Failed to connect to ${config.gear.wsProvider}`);
+    connectionStatus = false;
+    logger.error(`Failed to connect to ${providerAdd}`);
+    indexer.stop();
+    await retryConnectionToNode(api, indexer, cb);
   }
   await api.isReady;
   const genesis = api.genesisHash.toHex();
 
   api.on('disconnected', () => {
-    reconnect(api, genesis, indexer, cb);
+    connectionStatus = false;
+    indexer.stop();
+    genesis && cb(RMQServiceActions.DELETE, genesis);
+    retryConnectionToNode(api, indexer, cb);
   });
 
-  reconnectionsCounter = 0;
-
-  indexer.run(api);
-  cb(RMQServiceActions.ADD, genesis);
 
+  reconnectionsCounter = 0;
+  connectionStatus = true;
+  connectionStatus && api && await indexer.run(api);
+  connectionStatus && cb(RMQServiceActions.ADD, genesis);
   logger.info(`⚙️ Connected to ${api.runtimeChain} with genesis ${genesis}`);
   changeStatus('gear');
 }
 
-async function reconnect(api: GearApi, genesis: string, indexer: GearIndexer, cb: GenesisCb) {
+async function retryConnectionToNode(api: GearApi, indexer: GearIndexer, cb: GenesisCb) {
+  if (addresses.length === 0) throw new Error('️ 📡 Unable to connect node providers 🔴');
+
+  if(connectionStatus) return;
+
+  for(let i = 0; i <= addresses.length; i + 1){
```

Comment:
Pls try to set up linter
```suggestion
  for (let i = 0; i <= addresses.length; i + 1) {
```